src/cascadia/TerminalApp/TerminalTab.cpp

```diff
@@ -690,37 +709,39 @@ namespace winrt::TerminalApp::implementation
     // - <none>
     void TerminalTab::_UpdateProgressState()
     {
-        if (const auto& activeControl{ GetActiveTerminalControl() })
+        const auto state{ GetCombinedTaskbarState() };

```

Comment:
gentle nit: we could compute the "max state" one time and store it, here, because this is the listener for all progress state changes (right?). That would make the call in TerminalPage free (it wouldn't need to `min_element` the group of winrt objects)